enhancements/api-review/add-new-CRD-ImageContentPolicy(ICP)-to-config.openshift.io.md

```diff
@@ -0,0 +1,231 @@
+- [openshift/machine-config-operator](https://github.com/openshift/machine-config-operator): MCO needs watch for the ImageContentPolicy.
+
+#### Constraints Need Discussion
+
+1. The impact on ImageStream and `oc` command is undefined.
+2. Design of the rejction of conflicting `allowMirrorByTags` values for the same `source` of multiple indenpendant CRs.
+3. The merge order of mirrors for the same source.<br/>
+   Current order is preserving the relative order of the mirrors, if possible.
+   For example, if policy A has mirrors `a, b, c` and policy B has mirrors `c, d, e`, the
+   mirrors will be used in the order `a, b, c, d, e`.  If the orders of mirror entries conflict
+   (e.g. `a, b` vs. `b, a`) the configuration is not rejected but the resulting order is unspecified.
+
+
+
+### Risks and Mitigations
+
+Pulling images from mirror registries without the digest specifications could lead to returning different image version from different registry if the image tag mapping is out of sync. But the OpenShift core required image using digests to avoid different versions won't consume this feature at all, so it is not exposed to the risks that anyone who actually uses the feature will be exposed to.
+
+## Design Details
+
+### Test Plan
+
+Update the tests that are currently in the MCO to verify that `mirror-by-digest-only` not have been set when spec `allowMirrorByTags: true` is specified in the ImageContentPolicy.
+
+### Graduation Criteria
+
+GA in openshift will be looked at to
+determine graduation.
+
+#### Dev Preview -> Tech Preview
+
+- Ability to utilize the enhancement end to end
+- End user documentation, relative API stability
+- Sufficient test coverage
+- Gather feedback from users rather than just developers
+- Enumerate service level indicators (SLIs), expose SLIs as metrics
+- Write symptoms-based alerts for the component(s)
+
+#### Tech Preview -> GA
+
+- More testing (upgrade, downgrade, scale)
+- Sufficient time for feedback
+- Available by default
+- Backhaul SLI telemetry
+- Document SLOs for the component
+- Conduct load testing
+
+#### Removing a deprecated feature
+
+- The CRD ImageContentSourcePolicy(ICSP) will be removed. A Jira card [OCPNODE-717](https://issues.redhat.com/browse/OCPNODE-717) was created to record the upgrade of ImageContenPolicy(ICP) . The [repositories](https://docs.google.com/document/d/11FJPpIYAQLj5EcYiJtbi_bNkAcJa2hCLV63WvoDsrcQ/edit) currently rely on operator/v1alpha1 ImageContentSourcePolicy(ICSP) will be migrated to config/v1 ImageContentPolicy.
+
+### Upgrade / Downgrade Strategy
+
+Upgrades should not be affect. But the downgrade will be affected.
+Users upgraded and configured the `ImageContentPolicy` spec will presumably have their CRI-O configurations clobbered and break their tag-mirroring-dependent workflows after they downgrade to a version that lacks support for the `ImageContentPolicy`.
```

Comment:
It’s not just tag-mirroring-dependent, it’s _all_ mirrors configured via ICP, isn’t it?

(I don’t think that’s a problem, in itself.)